### DISCUSSION TITLE
Capitalise.ai Timed Out Error

### DISCUSSION THREAD

**Discussion by IRONJJ** [soranoo/TradingView-Free-Webhook-Alerts#41] [Q&A] [answered]
I am constantly getting the following error: 

2023-06-06T22:07:53.590Z	fd792c4a-7229-4699-8a8c-4b4588f71824	TRACE	Error
    at NanoWorker.kill (file:///var/task/nano_worker.mjs:135:19)
    at NanoWorker.start (file:///var/task/nano_worker.mjs:49:18)
    at NanoWorker.restart (file:///var/task/nano_worker.mjs:143:23)
    at Object.restart (file:///var/task/nano_worker.mjs:162:16)
    at Runner.run (file:///var/task/lambda_handler.mjs:703:32)
    at async Runtime.handler (file:///var/task/lambda_handler.mjs:914:22)

and 

TIMEOUT
TIMEOUT

DETAILS
TIMEOUT: TIMEOUT
    at Timeout._onTimeout (file:///var/task/lambda_handler.mjs:884:23)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
    
Any advice on how to fix this? 
And does this script allow you to send specific alert URL's to a webhook? 
The platform I am using gives the webhook and alert ID separately. So I send the individual alert ID in full JSON format to the Webhook which is the link they provide.

**Comment by soranoo** [comment]
Would you mind providing some screenshots about the timed-out issue? I never see this timed-out error in the past.

Regarding the second question, do you mean you have to send the alert ID to the system first after that the signal can be sent?

**Comment by IRONJJ** [comment]
For the first question: I will send screenshots when I am back home. But in
the meantime.

Regarding the 2nd question. I input the Alert ID's for long/short/exits on
tradingview in full JSON format. So if the strategy runs it will trigger
the relevant alert ID to long/short/exit. So yes. I have to send the alert
ID to the platform in order for it to send the signal to the crypto
platform and open/close the trades. According to their guide they provide a
webhook url and they provide an alert ID in already JSON format. Which you
can just copy and paste. Every webhook you create has its own unique ID.

**Comment by soranoo** [comment]
Yes, you can. Any type of webhook that accepts JSON is ok

**Comment by IRONJJ** [comment]
Thank you very much.

Which screenshots should I take for you? About the errors or the entire
code?

**Reply by soranoo** [reply]
The errors(with the whole application window) only if you hadn't modified the code.

**Reply by IRONJJ** [reply]
![Screenshot (41)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/60feee83-598b-4c6c-bc85-1627abb4e585)
![Screenshot (42)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/72b8f849-82aa-4f38-a50e-d9e59f9eb7b6)
![Screenshot (44)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/ce7f04b3-a4f2-42f3-bec6-8c72b51187b6)
![Screenshot (45)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/08c0cca4-2ffb-486b-8ece-e9b52c020801)
![Screenshot (46)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/188889af-bc00-4161-82f1-0386ad5873e7)

![Screenshot (47)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/95f3aa34-a66e-40b5-a51e-af0598e9a9e1)
![Screenshot (48)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/52a2e653-07bc-4c8d-9748-1da015d8f12e)
![Screenshot (49)](https://github.com/soranoo/TradingView-Free-Webhook-Alerts/assets/30935131/279f7da8-dbed-40a4-84c9-568fd4f9b8c3)

I haven't modified the code. I only added the webhook.site link in order to test if it was maybe the link but it seems not.

**Reply by IRONJJ** [reply]
After testing, it seems like it is the webhook URL of the platform I am using. Do you know what the cause might be? Or how I can fix it? The URL is correct.

**Reply by soranoo** [reply]
There is something wrong with https://tvwebhook.capitalise.ai.

**Reply by soranoo** [reply]
I cann't make any request to it.

**Reply by IRONJJ** [reply]
Might it be that they are blocking other sources except for tradingview?

**Reply by soranoo** [reply]
Yes, they may set restrictions on other IPs which not on the TV IP list.

Ref: https://www.tradingview.com/support/solutions/43000529348-about-webhooks/

**Reply by IRONJJ** [reply]
I contacted them. They Only allow alerts from Tradingview directly.

**Reply by IRONJJ** [reply]
Is there a way to maybe route the webhook alerts via those IP's? Sorry if it is a dumb question I am not very informed when it comes to coding.

**Reply by soranoo** [reply]
I don't think it is possible.

**Reply by IRONJJ** [reply]
Thank you for the assistance :)

**Comment by soranoo** [comment, marked as answer]
For someone who is looking for the answer.

[Capitalise.ai](https://capitalise.ai/) seems set restrictions on other IPs which not on the TradingView IP list so the timed out error comes out.

To sum up, this program is **not suitable** for [Capitalise.ai](https://capitalise.ai/).

Ref: https://www.tradingview.com/support/solutions/43000529348-about-webhooks/
